File: Java/프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[2];
        int left = 0;
        int right = 0;
        int sum = 0;
        int pre = sequence.length;

        for (right = 0; right < sequence.length; right++) {
            sum += sequence[right];
            while (right < sequence.length && sum > k) {
                sum -= sequence[left];
                left++;
            }
            if (sum == k && right - left < pre) {
                pre = right - left;
                answer[0] = left;
                answer[1] = right;
            }
        }

        return answer;
    }
}

File: Python/프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    n = len(sequence)
    answer = []
    gap = n + 1
    arr_sum = [0] * (n + 1)
    arr_sum[1] = sequence[0]

    for i in range(n):
        arr_sum[i + 1] = arr_sum[i] + sequence[i]

    s, e = 0, 1
    while e < n + 1:
        if arr_sum[e] - arr_sum[s] == k:
            if e - s < gap:
                gap = e - s
                answer = [s, e - 1]
            s += 1
            e += 1
        elif arr_sum[e] - arr_sum[s] > k:
            s += 1
        else:
            e += 1

    return answer